modules/imap/functions.php

```diff
@@ -212,122 +212,121 @@ function format_imap_message_list($msg_list, $output_module, $parent_list=false,
     foreach($msg_list as $msg) {
         $row_class = 'email';
         $icon = 'env_open';
-        if (!$parent_list) {
-            $parent_value = sprintf('imap_%s_%s', $msg['server_id'], $msg['folder']);
-        }
-        else {
-            $parent_value = $parent_list;
-        }
-        $id = sprintf("imap_%s_%s_%s", $msg['server_id'], $msg['uid'], $msg['folder']);
-        if (!trim($msg['subject'])) {
-            $msg['subject'] = '[No Subject]';
-        }
-        $subject = $msg['subject'];
-        $preview_msg = "";
-        if (isset($msg['preview_msg'])) {
-            $preview_msg = $msg['preview_msg'];
-        }
-       
-        if ($parent_list == 'sent') {
-            $icon = 'sent';
-            $from = $msg['to'];
-        }
-        else {
-            $from = $msg['from'];
-        }
-        $from = format_imap_from_fld($from);
-        $nofrom = '';
-        if (!trim($from)) {
-            $from = '[No From]';
-            $nofrom = ' nofrom';
-        }
-        $is_snoozed = !empty($msg['x_snoozed']) && hex2bin($msg['folder']) == 'Snoozed';
-        if ($is_snoozed) {
-            $snooze_header = parse_snooze_header('X-Snoozed: '.$msg['x_snoozed']);
-            $date = $snooze_header['until'];
-            $timestamp = strtotime($date);
-        } else {
-            if ($list_sort == 'date') {
-                $date_field = 'date';
+        if ($msg) {
```

Comment:
Please return this function to the previous state - we shouldn't have empty messages in the list.